### PR TITLE
Added support Nyx Rev.2 soldering

### PR DESCRIPTION
**Pull request by HorrorTroll**
## Description

Added support Nyx Rev.2 soldering.
This PR need to be waiting for #21143 merged first.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by HorrorTroll** [comment]
Ready for reviewing now

**Review by drashna — APPROVED** [review]

File: keyboards/horrortroll/nyx/rev2/solder/config.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* RGB Matrix config */
+#define RGB_MATRIX_LED_COUNT 19
+#define RGB_MATRIX_DEFAULT_VAL RGB_MATRIX_MAXIMUM_BRIGHTNESS
```
Comment (fauxpark) [inline keyboards/horrortroll/nyx/rev2/solder/config.h:21]: These can be removed and this file deleted; `RGB_MATRIX_DEFAULT_VAL` should be set in info.json.

File: keyboards/horrortroll/nyx/rev2/solder/readme.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Nyx (Solder)
+
+![Nyx](https://i.imgur.com/7ccW7wPh.png)
```
Comment (fauxpark) [inline keyboards/horrortroll/nyx/rev2/solder/readme.md:3]: Dead link.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/horrortroll/nyx/rev2/solder/readme.md (hunk 2 above)
Comment (HorrorTroll) [reply keyboards/horrortroll/nyx/rev2/solder/readme.md:3]: image

That weird, the link still work on my side here. Can you re-check again?

Comment (fauxpark) [reply keyboards/horrortroll/nyx/rev2/solder/readme.md:3]: Still doesn't work when I click on it or paste it directly, although Camo seems to have no problems displaying it. 🤷

File: keyboards/horrortroll/nyx/rev2/solder/config.h (hunk 1 above)
Comment (fauxpark) [reply keyboards/horrortroll/nyx/rev2/solder/config.h:21]: Still outstanding.

Comment (HorrorTroll) [reply keyboards/horrortroll/nyx/rev2/solder/config.h]: Thanks, I've push a new commit to resolved this.

File: keyboards/horrortroll/nyx/rev2/solder/rules.mk
Comment (fauxpark) [inline keyboards/horrortroll/nyx/rev2/solder/rules.mk:1]: This file can be deleted.

File: keyboards/horrortroll/nyx/rev2/solder/info.json
Comment (fauxpark) [inline keyboards/horrortroll/nyx/rev2/solder/info.json:1]: This file should be renamed to `keyboard.json`.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/horrortroll/nyx/rev2/solder/info.json
Comment (HorrorTroll) [reply keyboards/horrortroll/nyx/rev2/solder/info.json:1]: Thanks for reviewing, I've just push a new update to resolve those issue. Feel free to check again if you have any time.

**Comment by tzarc** [comment]
As of August 26, 2024, `qmk/qmk_firmware` is no longer accepting VIA-enabled keymaps as these have now transitioned to a repository under the VIA team's control.

As you've submitted a PR containing `via` or VIA-enabled keymap(s), this is your notice that they should be removed from this PR. You should now submit a secondary PR to the [VIA QMK Userspace](https://github.com/the-via/qmk_userspace_via) repository with your associated `via` or VIA-enabled keymaps instead.

File: keyboards/horrortroll/nyx/rev2/solder/keymaps/via/keymap.c
Comment (fauxpark) [inline keyboards/horrortroll/nyx/rev2/solder/keymaps/via/keymap.c:1]: Via keymap needs to be removed.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**